consensus/polybft/consensus_runtime.go

```diff
@@ -253,8 +258,9 @@ func (c *consensusRuntime) populateFsmIfBridgeEnabled(
 func (c *consensusRuntime) FSM() (*fsm, error) {
 	// figure out the parent. At this point this peer has done its best to sync up
 	// to the head of their remote peers.
-	parent := c.lastBuiltBlock
-	epoch := c.getEpoch()
+	c.lock.RLock()
```

Comment:
As per [comment](https://github.com/0xPolygon/polygon-edge/pull/849/files#r1012728673), I think you do not need a lock around `lastBuiltBlock`. You do need a lock on getting `epochMetadata`, but you have `getEpoch` function you can use here.

Reply:
Could we introduce thread safe methods rather then explicit call on mutexes?

Reply:
Currently we have one get method for each field. 

if we call them from this place one after another than it is possible to retrieve (for example) epoch from previous state, and last block from the newest.

Technically these two fields should be in some other struct but that is a lot of work right now I think.

Is it ok to create another method which will return both of those fields together like `getEpochAndLastBuiltBlock() (uint64, *epochMetaData)`?

Reply:
the last one works for me, as far as we don't change *epochMetaData.

Reply:
or make a small struct and store it as atomic in future with atomic.Pointer[*T]

Reply:
Fixed, @JekaMas please check https://github.com/0xPolygon/polygon-edge/pull/849/commits/55c2213f556ab9db005a6eb60203d4551feab5a2